File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/IUnitOfWorkManager.java
package com.ueueo.uow;

import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

/**
 * @author Lee
 * @date 2022-05-19 20:32
 */
public interface IUnitOfWorkManager {
    @Nullable
    IUnitOfWork getCurrent();

    @NonNull
    IUnitOfWork begin(@NonNull AbpUnitOfWorkOptions options, Boolean requiresNew);

    @NonNull
    IUnitOfWork reserve(@NonNull String reservationName, Boolean requiresNew);

    void beginReserved(@NonNull String reservationName, @NonNull AbpUnitOfWorkOptions options);

    boolean tryBeginReserved(@NonNull String reservationName, @NonNull AbpUnitOfWorkOptions options);
}

File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/IUnitOfWorkManager.cs
using JetBrains.Annotations;

namespace Volo.Abp.Uow;

public interface IUnitOfWorkManager
{
    [CanBeNull]
    IUnitOfWork Current;//  { get; }

    [NotNull]
    IUnitOfWork Begin(@NonNull AbpUnitOfWorkOptions options, boolean requiresNew = false);

    [NotNull]
    IUnitOfWork Reserve(@NonNull String reservationName, boolean requiresNew = false);

    void BeginReserved(@NonNull String reservationName, @NonNull AbpUnitOfWorkOptions options);

    boolean TryBeginReserved(@NonNull String reservationName, @NonNull AbpUnitOfWorkOptions options);
}
